python-checks/src/main/java/org/sonar/python/checks/hotspots/UnsafeHttpMethodsCheck.java

```diff
@@ -121,4 +135,31 @@ private static boolean isDjangoView(FunctionDef functionDef) {
       .filter(FunctionSymbolImpl::isDjangoView)
       .isPresent();
   }
+
+  private Optional<CallExpression> getFlaskViewDecorator(FunctionDef functionDef) {
+    return functionDef.decorators().stream()
+      .map(Decorator::expression)
+      .filter(expression -> expression.is(CALL_EXPR))
+      .map(CallExpression.class::cast)
+      .filter(this::isFlaskRouteDecorator)
+      .findFirst();
+  }
+
+  private boolean isFlaskRouteDecorator(CallExpression callExpression) {
+    Symbol calleeSymbol = callExpression.calleeSymbol();
+    if (calleeSymbol == null) {
+      return false;
+    }
+    if (calleeSymbol.fullyQualifiedName() == null) {
+      return isFlaskImported && calleeSymbol.name().equals("route");
```

Comment:
I think I understand why we do that and I don't really like it.
If that's possible, I'd prefer a cleaner solution: maybe we could adjust how we handle module variables.
We can talk about it.

Reply:
After discussion, I'll change the implementation to avoid keeping a state, however keeping this "hack" to infer if a decorator defines a flask route.
I also created [SONARPY-834](https://jira.sonarsource.com/browse/SONARPY-834) which enables type inference at module level and would allow us to have a cleaner implementation of this.